certs/pki/vault.go

```diff
@@ -179,15 +148,18 @@ func (p *pkiAgent) Read(serial string) (Cert, error) {
 }
 
 func (p *pkiAgent) Revoke(serial string) (time.Time, error) {
+	// We aren't using p.client.Logical().Delete() since it is making a DELETE request to vault
+	// yet for certificate revoking vault needs a POST request to the same URL
+	// This is a library issue and will be solved once a POST request is added to p.client.Logical()
+	// as the current alternatives either use a PUT or DELETE request
 	cReq := certRevokeReq{
-		SerialNumber: serial,
+		SerialNumber: strings.Replace(serial, ":", "-", -1),
```

Comment:
Can you add a quick comment explaining this line?

Reply:
Use `http.MethodPost` instead of the string in:
```go
r := p.client.NewRequest("POST", p.revokeURL)
```

Reply:
updated